src/app/ReadHandler.cpp

```diff
@@ -252,6 +257,7 @@ CHIP_ERROR ReadHandler::SendReportData(System::PacketBufferHandle && aPayload, b
     }
     if (!aMoreChunks)
     {
+        mLastReportTick = mCurrentReportTick;
```

Comment:
This should actually be snapping to the timestamp of the CURRENT report, not the first report in the CURRENT reporting series/train.